### PR TITLE
fix(pie-notification): DSW-000 make sure custom and default icons work as expected

### PR DESCRIPTION
**Pull request by dandel10n**
## Describe your changes (can list changeset entries if preferable)

---
"@justeattakeaway/pie-notification": patch
---

[Fixed] - make sure custom and default icons work as expected.
[Fixed] - make `isDismissible` false by default
[Fixed] - use storybook mapping to render different slot content

## Author Checklist (complete before requesting a review)
- [x] I have performed a self-review of my code
- [x] I have reviewed the `PIE Storybook`/`PIE Docs` PR preview
- [x] I have reviewed visual test updates properly before approving

### Reviewer 1 - @raoufswe 
- [x] I have reviewed the `PIE Storybook` PR preview
- [x] If there are visual test updates, I have reviewed them

### Reviewer 2 - @xander-marjoram 
- [x] I have reviewed the `PIE Storybook` PR preview
- [x] If there are visual test updates, I have reviewed them

### CODE REVIEW

**Review by raoufswe — APPROVED** [review]

File: apps/pie-storybook/stories/pie-notification.stories.ts [hunk 1]
```diff
@@ -141,9 +155,16 @@ const Template : TemplateFunction<NotificationProps> = ({
     supportingAction,
     hasStackedActions,
     slot,
+    iconSlot,
 }) => {
     const shouldShowPlaceholderIcon = variant && ['neutral', 'neutral-alternative'].includes(variant);
```
Comment (xander-marjoram) [inline apps/pie-storybook/stories/pie-notification.stories.ts:160]: Is this used anymore?

File: packages/components/pie-notification/src/index.ts [hunk 2]
```diff
@@ -42,8 +42,13 @@ export * from './defs';
  * @event {CustomEvent} pie-notification-supporting-action-click - When the notification supporting action is clicked.
  * @event {CustomEvent} pie-notification-close - When the notification is closed.
  * @event {CustomEvent} pie-notification-open - When the notification is opened.
+ * @slot - Default slot
+ * @slot icon - The icon slot
  */
 export class PieNotification extends LitElement implements NotificationProps {
+    @state()
+        hasExternalIcon = false;
```
Comment (xander-marjoram) [inline packages/components/pie-notification/src/index.ts:50]: Is this for internal use only? If so, we should probably mark is as `private` (and fix the indentation 😄)

File: apps/pie-storybook/stories/pie-notification.stories.ts (hunk 1 above)
Comment (raoufswe) [reply apps/pie-storybook/stories/pie-notification.stories.ts]: nope

File: packages/components/pie-notification/src/index.ts [hunk 3]
```diff
@@ -160,11 +170,9 @@ export class PieNotification extends LitElement implements NotificationProps {
      * @private
      */
     private renderIcon (): TemplateResult | typeof nothing {
-        const hasExternalIcon = this._iconSlot.length > 0;
-
         return html`
-                ${!hasExternalIcon ? this.getDefaultVariantIcon() : nothing}
-                <slot name="icon"></slot>
+                ${!this.hasExternalIcon ? this.getDefaultVariantIcon() : nothing}
+                <slot @slotchange=${this.handleSlotchange} name="icon"></slot>
```
Comment (xander-marjoram) [inline packages/components/pie-notification/src/index.ts:175]: I've just realised we might be able to do something different and much simpler...
https://lit.dev/docs/components/shadow-dom/#fallback

I think it's worth trying before we merge this.

```suggestion
                <slot name="icon">${this.getDefaultVariantIcon()}</slot>
```

Comment (raoufswe) [reply packages/components/pie-notification/src/index.ts]: looks like it works, great finding 🔥

**Review by raoufswe — APPROVED** [review]

**Review by raoufswe — APPROVED** [review]

**Review by xander-marjoram — APPROVED** [review]

**Pull request merged**